app/Http/Controllers/GalaxyController.php

```diff
@@ -438,4 +440,26 @@ private function getPlanetActivityStatus(PlanetService $planet): array
 
         return $result;
     }
+
+    /**
+     * Calculate the number of colonized slots in the galaxy.
+     *
+     * @param array<int, array<string, mixed>> $galaxyContent
+     * @return int
+     */
+    private function calculateColonizedSlots(array $galaxyContent): int
+    {
+        $slotsColonized = 0;
+
+        foreach ($galaxyContent as $record) {
+            if (isset($record['planets']) && !empty($record['planets'])) {
+                $activePlanets = array_filter($record['planets'], function ($planet) {
+                    return (!isset($planet['isDestroyed']) || !$planet['isDestroyed']) && isset($planet['playerId']);
+                });
+                $slotsColonized += count($activePlanets);
+            }
```

Comment:
Hi @gabrieltsants,

I just tested the new changes and functionality wise its looking good now. I'm getting accurate counts with this version so that seems to be working well 👍.

For the changes I have a small tip, and that is to use the `!empty()` check instead of the `isset()` as the former already checks if the key in the array is set or not. So in this case the `isset()` is not necessary and can be removed. This will make the code a little bit cleaner and easier to read.

Concrete if you could make the code like this I think it should still work and is more concise:

```
            if (!empty($record['planets'])) {
                $activePlanets = array_filter($record['planets'], function ($planet) {
                    return empty($planet['isDestroyed']) && !empty($planet['playerId']);
                });
                $slotsColonized += count($activePlanets);
            }
```

I have not tested it yet but think it would work. Can you see if you can refactor this method?

Again functionality wise it looks good to me so with this change I'll be happy to merge it. Thanks again for your time! 🚀

Reply:
Hi @lanedirt,

I have made the changes as per your suggestion and pushed the updated code. Please feel free to test it again, and let me know if there’s anything else needed.

Thank you for your guidance, and I remain at your disposal for any further adjustments 😁